components/layout_2020/construct_modern.rs

```diff
@@ -103,31 +97,36 @@ where
     }
 }
 
-/// <https://drafts.csswg.org/css-text/#white-space>
-fn is_only_document_white_space<Node>(run: &FlexTextRun<'_, Node>) -> bool {
-    // FIXME: is this the right definition? See
-    // https://github.com/w3c/csswg-drafts/issues/5146
-    // https://github.com/w3c/csswg-drafts/issues/5147
-    run.text
-        .bytes()
-        .all(|byte| matches!(byte, b' ' | b'\n' | b'\t'))
-}
-
-impl<'a, 'dom, Node: 'dom> FlexContainerBuilder<'a, 'dom, Node>
+impl<'a, 'dom, Node: 'dom> ModernContainerBuilder<'a, 'dom, Node>
 where
     Node: NodeExt<'dom>,
 {
+    pub fn new(
+        context: &'a LayoutContext<'a>,
+        info: &'a NodeAndStyleInfo<Node>,
+        text_decoration_line: TextDecorationLine,
+    ) -> Self {
+        ModernContainerBuilder {
+            context,
+            info,
+            text_decoration_line,
+            contiguous_text_runs: Vec::new(),
+            jobs: Vec::new(),
+            has_text_runs: false,
+        }
+    }
+
     fn wrap_any_text_in_anonymous_block_container(&mut self) {
         let runs = std::mem::take(&mut self.contiguous_text_runs);
-        if runs.iter().all(is_only_document_white_space) {
+        if runs.iter().all(|r| is_only_document_white_space(&r.text)) {
```

Comment:
Then this can just be:

```suggestion
        if runs.iter().all(FlexItemTextRun::is_only_document_white_space) {
```